independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/ClientBuilderImpl.java

```diff
@@ -174,10 +187,41 @@ public ClientImpl build() {
         }
 
         if (proxyHost != null) {
-            options.setProxyOptions(
-                    new ProxyOptions()
-                            .setHost(proxyHost)
-                            .setPort(proxyPort));
+            ProxyOptions proxyOptions = new ProxyOptions()
+                    .setHost(proxyHost)
+                    .setPort(proxyPort);
+            if (proxyPassword != null && !proxyPassword.isBlank()) {
+                proxyOptions.setPassword(proxyPassword);
+            }
+            if (proxyUser != null && !proxyUser.isBlank()) {
+                proxyOptions.setUsername(proxyUser);
+            }
+            options.setProxyOptions(proxyOptions);
+        } else {
+            String proxyHost = options.isSsl()
+                    ? System.getProperty("https.proxyHost", "none")
+                    : System.getProperty("http.proxyHost", "none");
+            String proxyPortAsString = options.isSsl()
+                    ? System.getProperty("https.proxyPort", "443")
+                    : System.getProperty("http.proxyPort", "80");
+            int proxyPort = Integer.parseInt(proxyPortAsString);
+
+            if (!"none".equals(proxyHost)) {
```

Comment:
We don't support `http.nonProxyHosts`? We do in the non-reactive client (see https://github.com/quarkusio/quarkus/pull/19261/files).

Reply:
good catch, I didn't have a todo for it in the code and forgot about it during the Xmas break :(
I'll try to add it soon. Unfortunately, Vert.x doesn't have a property for it.